execution_engine/src/storage/global_state/scratch.rs

```diff
@@ -0,0 +1,495 @@
+            },
+            TestPair {
+                key: Key::Account(AccountHash::new([2_u8; 32])),
+                value: StoredValue::CLValue(CLValue::from_t(2_i32).unwrap()),
+            },
+        ]
+    }
+
+    fn create_test_pairs_updated() -> [TestPair; 3] {
+        [
+            TestPair {
+                key: Key::Account(AccountHash::new([1u8; 32])),
+                value: StoredValue::CLValue(CLValue::from_t("one".to_string()).unwrap()),
+            },
+            TestPair {
+                key: Key::Account(AccountHash::new([2u8; 32])),
+                value: StoredValue::CLValue(CLValue::from_t("two".to_string()).unwrap()),
+            },
+            TestPair {
+                key: Key::Account(AccountHash::new([3u8; 32])),
+                value: StoredValue::CLValue(CLValue::from_t(3_i32).unwrap()),
+            },
+        ]
+    }
+
+    struct TestState {
+        state: LmdbGlobalState,
+        root_hash: Digest,
+    }
+
+    fn create_test_state() -> TestState {
+        let correlation_id = CorrelationId::new();
+        let temp_dir = tempdir().unwrap();
+        let environment = Arc::new(
+            LmdbEnvironment::new(
+                &temp_dir.path().to_path_buf(),
+                DEFAULT_TEST_MAX_DB_SIZE,
+                DEFAULT_TEST_MAX_READERS,
+                true,
+            )
+            .unwrap(),
+        );
+        let trie_store =
+            Arc::new(LmdbTrieStore::new(&environment, None, DatabaseFlags::empty()).unwrap());
+
+        let state = LmdbGlobalState::empty(environment, trie_store).unwrap();
+        let mut current_root = state.empty_root_hash;
+        {
+            let mut txn = state.environment.create_read_write_txn().unwrap();
+
+            for TestPair { key, value } in &create_test_pairs() {
+                match write::<_, _, _, LmdbTrieStore, error::Error>(
+                    correlation_id,
+                    &mut txn,
+                    &state.trie_store,
+                    &current_root,
+                    key,
+                    value,
+                )
+                .unwrap()
+                {
+                    WriteResult::Written(root_hash) => {
+                        current_root = root_hash;
+                    }
+                    WriteResult::AlreadyExists => (),
+                    WriteResult::RootNotFound => {
+                        panic!("LmdbWithCacheGlobalState has invalid root")
+                    }
+                }
+            }
+
+            txn.commit().unwrap();
+        }
+        TestState {
+            state,
+            root_hash: current_root,
+        }
+    }
+
+    #[test]
+    fn checkout_fails_if_unknown_hash_is_given() {
+        let TestState { state, .. } = create_test_state();
+        let fake_hash: Digest = Digest::hash(&[1u8; 32]);
+        let result = state.checkout(fake_hash).unwrap();
+        assert!(result.is_none());
+    }
```

Comment:
This test doesn't check anything new added in this PR as far as I can tell - the `TestState.state` is an `LmdbGlobalState` so the `state.checkout` unit tests the LMDB. Let's either move it to the correct module or delete it from here as it's creating a false sense of security about the `ScratchState`.

Reply:
Deleted it, thanks